### PR TITLE
Fp/run benchmark on more platforms 2118

### PR DESCRIPTION
**Pull request by papafe**
Added benchmarks on Linux and MacOS

### CODE REVIEW

**Comment by papafe** [comment]
Benchmarks now run on both MacOS and Linux with a .NET 5 target.
Eventually one could "merge" some jobs (especially Linux and MacOS ones), in order to have less repetitive code, but I think that it would just make things less readable and more complicated to modify.

**Review by nirinchev — COMMENTED** [review]
A few nits, looking good otherwise!

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        run:  |
+          dotnet run -p .\Tests\PerformanceTests\ run -c Release --framework net5.0 -- -f * --join
+          rename-item -path ".\BenchmarkDotNet.Artifacts\results\*.json" -newname "benchmark-output.json"
+      - name: Save artifacts
+        # Not mandatory, but maybe it's good to do it?
+        uses: actions/upload-artifact@v2
+        with:
+          name: benchmark-results-windows
+          path: .\BenchmarkDotNet.Artifacts\results
+      # Download previous benchmark result from cache (if exists)
```
[the remark this answers is not in the document]
Comment (papafe) [reply .github/workflows/benchmark.yml:103]: This comment should actually be some lines down, it explains what `github-action-benchmark` does, but probably it's not necessary.

**Comment by papafe** [comment]
Finished corrections according to comments. 
As an additional note, for now the Windows branch is the only one pushing to to Github pages for the graph. At the moment we don't have a clear idea about how/what to show there, so I thought it was ok to leave it like this

**Review by nirinchev — APPROVED** [review]

**Pull request merged**